src/pyenphase/envoy.py

```diff
@@ -258,10 +261,12 @@ async def _request(
             )
         # show all responses centrally when in debug
         if debugon:
+            request_end = time.monotonic()
             content_type = response.headers.get("content-type")
             content = response.content
             _LOGGER.debug(
-                "Request reply from %s status %s: %s %s",
```

Comment:
The debug could be guarded as well so it doesn’t have to check twice

Reply:
Not sure I see the guard here https://github.com/pyenphase/pyenphase/blob/4447ce76ea6fb23ac99f8df4463bac1edc8abd61/src/pyenphase/envoy.py#L256-L276

Using below still requires a test I'd think. And returning response twice. So I'm probably missing the real point for the guard you have in mind.
```py
        if not debugon:
            return response
        request_end = time.monotonic()
        content_type = response.headers.get("content-type")
        content = response.content
        _LOGGER.debug(
                "Request reply in %s sec from %s status %s: %s %s",
                round(request_end - request_start, 1),
                url,
                status_code,
                content_type,
                content,
        )

        return response
```

Firmware.py would be a case for a guard I guess.
https://github.com/pyenphase/pyenphase/blob/4447ce76ea6fb23ac99f8df4463bac1edc8abd61/src/pyenphase/firmware.py#L83-L110

Reply:
Looks like the _LOGGER.debug is already guarded with `debugon` but I didn't scroll down far enough to see it on my mobile